bark/models/behavior/rule_based/mobil_behavior.cpp

```diff
@@ -49,47 +70,107 @@ BehaviorMobilRuleBased::ChooseLaneCorridor(
 
-  double acc_ego = 0., acc_change_ego = 0., acc_behind = 0.,
-         acc_change_behind = 0.;
+  double acc_ego = 0;
   std::pair<LaneCorridorInformation, bool> lci_has =
-      SelectLaneCorridor(lane_corr_infos, GetLaneCorridor());
-  if (std::get<1>(lci_has)) {
+      SelectLaneCorridor(lane_corr_infos, lane_corr);
+  if (lci_has.second) {
     LaneCorridorInformation lci = std::get<0>(lci_has);
-    if (lci.front.agent_info.first)
-      acc_ego = CalcIDMAcc(lci.front.rel_distance,
+    if (lci.front.agent_info.first) {
+      BARK_EXPECT_TRUE(lci.front.rel_distance >= 0);
+      acc_ego = CalcRawIDMAcc(lci.front.rel_distance,
                            GetVelocity(observed_world.GetEgoAgent()),
                            GetVelocity(lci.front.agent_info.first));
-    if (lci.rear.agent_info.first)
-      acc_behind = CalcIDMAcc(-lci.rear.rel_distance,
-                              GetVelocity(lci.rear.agent_info.first),
-                              GetVelocity(observed_world.GetEgoAgent()));
+    } else {
+      acc_ego = CalcLongRawAccWithoutLeader(
+          lane_corr, observed_world.CurrentEgoPosition(),
+          GetVelocity(observed_world.GetEgoAgent()));
+    }
+  } else {
+    LOG(INFO) << "Target Corridor not available to MOBIL" << std::endl;
+    // Vehicle cannot stay in current corridor anymore, we thus incentivize a
+    // lane change
+    acc_ego = BaseIDM::GetAccelerationLowerBound();
   }
+
   if (lane_corr_infos.size() > 0) {
     double max_advantage = -100.;
+    // use current lane corridor as default
     LaneCorridorPtr tmp_lane_corr = GetLaneCorridor();
     for (const auto& li : lane_corr_infos) {
-      if (li.front.agent_info.first)
-        acc_change_ego = CalcIDMAcc(li.front.rel_distance,
+      VLOG(2) << li << std::endl;
+      double acc_change_ego, acc_behind, acc_change_behind;
+      if (li.front.agent_info.first) {
+        BARK_EXPECT_TRUE(li.front.rel_distance >= 0);
+        acc_change_ego = CalcRawIDMAcc(li.front.rel_distance,
                                     GetVelocity(observed_world.GetEgoAgent()),
                                     GetVelocity(li.front.agent_info.first));
-      if (li.rear.agent_info.first)
-        acc_change_behind = CalcIDMAcc(
+      } else {
+        acc_change_ego = CalcLongRawAccWithoutLeader(
+            li.lane_corridor, observed_world.CurrentEgoPosition(),
+            GetVelocity(observed_world.GetEgoAgent()));
+      }
+      // Calculating acc of new follower before changing lane
+      if (li.rear.agent_info.first) {
+        if (li.front.agent_info.first) {
+          double distance = li.front.rel_distance - li.rear.rel_distance;
+          BARK_EXPECT_TRUE(distance >= 0);
+          acc_behind =
+              CalcRawIDMAcc(distance, GetVelocity(li.rear.agent_info.first),
+                         GetVelocity(li.front.agent_info.first));
+          VLOG(2) << "Distance rear vehicle to front before: " << distance
```

Comment:
How do you come up with the log level 2, if you use the model in forward simulation for MCTS, this will be very annoying. We should first define a mapping of loglevels to frequency of occurence of the event and then adjust all this stuff. For now, maybe switch to a higher level 5,6 ?

Reply:
It's defined with print everything at level xx and lower, so 2 will be printed only if the logging level is set to 2 or higher. the default log level is 0, so it should only print in MCTS, if you change the log level. However, I changed it to 4 to make some room.